tests/pytests/test_recovery.py

```diff
@@ -1552,3 +1553,181 @@ def test_recovery(self, use_server_send):
 
     def test_teardown(self):
         self.cleanup()
+
+
+@pytest.mark.usefixtures("servers")
+@pytest.mark.incremental
+class TestRecoveryDCWithUncommitted:
+    """This test cover the behaviour with uncommitted keys in safe and unsafe modes."""
+
+    timestamp_new = elliptics.Time.now()
+    timestamp_fresh = elliptics.Time(timestamp_new.tsec - 3600, timestamp_new.tnsec)
+    timestamp_old = elliptics.Time(timestamp_new.tsec - 25 * 3600, timestamp_new.tnsec)
+    timestamp_oldest = elliptics.Time(timestamp_new.tsec - 30 * 3600, timestamp_new.tnsec)
+    data = 'data'
+
+    """Helpers for creation and validation keys in group. There are such states:
+    * none:         missing key
+    * uncommitted:  started but not finalized write
+    * committed:    finalized write
+    """
+    def _create_none(self, key, group, timestamp):
+        pass
+
+    def _create_committed(self, key, group, timestamp):
+        scope.session.timestamp = timestamp
+        scope.session.groups = [group]
+        scope.session.write_data(key, self.data).wait()
+
+    def _create_uncommitted(self, key, group, timestamp):
+        scope.session.timestamp = timestamp
+        scope.session.groups = [group]
+        scope.session.write_prepare(key, self.data, 0, 1024).wait()
+
+    def _check_none(self, key, group, timestamp):
+        scope.session.groups = [group]
+        with pytest.raises(elliptics.NotFoundError):
+            scope.session.read_data(key).get()
```

Comment:
```suggestion
            scope.session.read_data(key).wait()
```

The result isn't needed.